contracts/cw-proposal-single/src/contract.rs

```diff
@@ -252,15 +256,39 @@ pub fn execute_execute(
         None => vec![],
     };
 
+    // TODO: This is not ready. Next Steps:
+    //
+    //  * Turn this into a query
+    //  * Return unauthorized if the query fails
+    //  * Remove the authorization from the "update" on the sub authorizations
+    //  * Add the update messages anyway but ignore the result
+    //  * Update all the tests to reflect the query changes
+    //  * Figure out a better way to add the auth to this proposal instead of using ReplaceOwner
+    //  * Write proper tests for the auth, and more realistic tests for this proposal where we don't fake the sender
+    //
+    let response = Response::default();
+    let response = if let Some(auths) = AUTHORIZATION_MODULE.may_load(deps.storage)? {
```

Comment:
are the performance implications of adding this query to every message negligible?